src/main/java/uk/gov/hmcts/ccd/domain/service/validate/CaseDataIssueLogger.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.ccd.domain.service.validate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.MapUtils;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class CaseDataIssueLogger {
+
+    private static final String EMPTY_COLLECTION_VALUE = "{}";
+
+    private CaseDataIssueLogger() {
+    }
+
+    public static List<String> logAnyDataIssuesIn(CaseDetails currentCaseDetails, CaseDetails newCaseDetails) {
+        try {
+            return logAnyUnexpectedEmptyCollectionIssueIn(currentCaseDetails, newCaseDetails);
+        } catch (Exception ex) {
+            log.error("Unexpected error while logging data issues in case details - ", ex);
+        }
+
+        return new ArrayList<>();
+    }
+
+    private static List<String> logAnyUnexpectedEmptyCollectionIssueIn(CaseDetails currentCaseDetails,
+                                                                 CaseDetails newCaseDetails) {
+        List<String> emptyCollectionsInCaseData = findNewlyAddedEmptyCollectionIn(
+            findAnyUnexpectedEmptyCollectionIssueIn(currentCaseDetails),
+            findAnyUnexpectedEmptyCollectionIssueIn(newCaseDetails));
+
+        if (!emptyCollectionsInCaseData.isEmpty()) {
+            log.debug("Case reference '{}' with state '{}' contains unexpected empty value in collection(s) '{}' "
+                    + "and the stacktrace - {}", newCaseDetails.getReference(), newCaseDetails.getState(),
+                emptyCollectionsInCaseData, new Exception().getStackTrace());
+        }
+
+        return emptyCollectionsInCaseData;
+    }
+
+    private static List<String> findNewlyAddedEmptyCollectionIn(List<String> emptyCollectionsInCurrentCaseData,
+                                                                List<String> emptyCollectionsInNewCaseData) {
+        return emptyCollectionsInNewCaseData.stream()
+            .filter(path -> !emptyCollectionsInCurrentCaseData.contains(path))
+            .collect(Collectors.toList());
+    }
+
+    private static List<String> findAnyUnexpectedEmptyCollectionIssueIn(CaseDetails caseDetails) {
+        List<String> emptyCollectionsInCaseData = new ArrayList<>();
+
+        if (caseDetails != null) {
+            Map<String, JsonNode> caseData = caseDetails.getData();
+            if (!MapUtils.isEmpty(caseData)) {
+                caseData.forEach((key, value) -> findEmptyCollectionsIn(value, key, emptyCollectionsInCaseData));
+            }
+        }
+
+        return emptyCollectionsInCaseData;
+    }
+
+    private static void findEmptyCollectionsIn(JsonNode node, String currentPath, List<String> emptyCollectionsInData) {
+        if (node == null) {
+            return;
+        }
+
```

Comment:
you can add more terminal conditions, like those for non-object value nodes.

Reply:
The second condition is the terminal condition for non-object value node. (node.isEmpty() && node.toString().trim().equals(EMPTY_COLLECTION_VALUE)